src/System.Data.SqlClient/tests/FunctionalTests/AmbientTransactionFailureTest.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using System.Transactions;
+using Xunit;
+
+namespace System.Data.SqlClient.Tests
+{
+    public class AmbientTransactionFailureTest
+    {
+
+        private static string servername = Guid.NewGuid().ToString();
+        private static string connectionStringWithEnlistAsDefault = $"Data Source={servername}; Integrated Security=true; Connect Timeout=1;";
+        private static string connectionStringWithEnlistOff = $"Data Source={servername}; Integrated Security=true; Connect Timeout=1;Enlist=False";
+
+        private static Action<string> ConnectToServer = (connectionString) =>
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+            }
+        };
+
+        private static Func<string, Task> ConnectToServerTask = (connectionString) =>
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                return connection.OpenAsync();
```

Comment:
Is this right? Asynchronously opening the connection, and immediately after initiating the operation disposing of the connection?